src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -294,19 +288,16 @@ private static boolean isAloneOnLine(Details details) {
     private static boolean isSingleLineBlock(Details details) {
         final DetailAST rcurly = details.rcurly;
         final DetailAST lcurly = details.lcurly;
-        final DetailAST nextToken = details.nextToken;
+        DetailAST nextToken = details.nextToken;
+        while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+            nextToken = Details.getNextToken(nextToken);
+        }
+        if (nextToken.getType() == TokenTypes.DO_WHILE) {
```

Comment:
@romani I am slightly confused, what does `else` and `do_while` have to do with determining if the current block is a single line?

Reply:
Examining the tests, there are only 2 instances this while loop changes the `nextToken`.

1)
https://github.com/checkstyle/checkstyle/blob/ecd1b65bd63546d1245b6415759c3c2d1e2b77a4/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAloneOrSingleline.java#L69
Next token is changed from else[69x30] to if[70x8].

https://github.com/checkstyle/checkstyle/blob/6a16499e5f5125ca7c6b2d9b554a7a44a65d5a2e/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAloneOrSingleLine2.java#L53
Next token is changed from else[53x28] to java[55x8].

Reply:
Maybe what is confusing me is that nextToken can be the next part of the chain (if-else) or it can be a separate statement (`if (true) { } int a = 0;`).
With the current code, `if (true) { return; } else { }` is considered a single line block, but as soon as you add something else to the end `if (true) { return; } else { } int a = 0;` it isn't a single long block anymore.

@romani Does this seem right?

Reply:
> Shouldn't this logic be placed inside getDetails instead of this section? Why is it only an override for isAloneOnLine?

I have no problem to do refactring.

> It seems to me, this checking of something on the end after the chain should be in a separate method check.

looks reasonable.